src/coreclr/jit/codegencommon.cpp

```diff
@@ -6966,6 +7727,22 @@ void CodeGen::genEstablishFramePointer(int delta, bool reportUnwindData)
         compiler->unwindSetFrameReg(REG_FPBASE, delta);
     }
 
+#elif defined(TARGET_LOONGARCH64)
+    if (delta == 0)
+    {
+        GetEmitter()->emitIns_R_R(INS_mov, EA_PTRSIZE, REG_FPBASE, REG_SPBASE);
+    }
+    else
+    {
+        assert((-2048 <= delta) && (delta < 2048));
```

Comment:
Why are we guaranteed that `delta` will fall in this range?

Reply:
Yes, you are right.
This seems a arbitrary decision.
Here is just move the sp to fp, if the outgoing space is large than 2047, we will first establish the sp-stack and move it to fp, then creating the outgoing space.

Now we didn't meet the outrange case here.